packages/SwingSet/src/kernel/vatManager/supervisor-subprocess-xsnap.js

```diff
@@ -0,0 +1,237 @@
+  // eslint-disable-next-line no-undef
+  issueCommand(Item.encode(['testLog', ...args]));
+}
+
+/**
+ * @param { (value: void) => void } f
+ * @param { string } errmsg
+ */
+function runAndWait(f, errmsg) {
+  Promise.resolve()
+    .then(f)
+    .then(undefined, err => workerLog(`doProcess: ${errmsg}:`, err));
+  return waitUntilQuiescent();
+}
+
+function makeWorker() {
+  /** @type { Record<string, (...args: unknown[]) => void> | null } */
+  let dispatch = null;
+
+  /** @type { (dr: Tagged, errmsg: string) => Promise<Tagged> } */
+  async function doProcess(dispatchRecord, errmsg) {
+    assert(dispatch);
+    const theDispatch = dispatch;
+    const [dispatchOp, ...dispatchArgs] = dispatchRecord;
+    assert(typeof dispatchOp === 'string');
+    workerLog(`runAndWait`);
+    await runAndWait(() => theDispatch[dispatchOp](...dispatchArgs), errmsg);
+    workerLog(`doProcess done`);
+    /** @type { Tagged } */
+    const vatDeliveryResults = harden(['ok']);
+    return vatDeliveryResults;
+  }
+
+  /** @type { (ts: unknown, msg: any) => Promise<Tagged> } */
+  function doMessage(targetSlot, msg) {
+    const errmsg = `vat[${targetSlot}].${msg.method} dispatch failed`;
+    return doProcess(
+      ['deliver', targetSlot, msg.method, msg.args, msg.result],
+      errmsg,
+    );
+  }
+
+  /** @type { (pv: unknown, msg: any) => Promise<Tagged> } */
+  function doNotify(primaryVpid, resolutions) {
+    const errmsg = `vat.promise[${primaryVpid}] failed`;
+    return doProcess(['notify', primaryVpid, resolutions], errmsg);
+  }
+
+  /**
+   * @param {unknown} vatID
+   * @param {unknown} bundle
+   * @param {unknown} vatParameters
+   * @param {unknown} virtualObjectCacheSize
+   * @returns { Promise<Tagged> }
+   */
+  async function setBundle(
+    vatID,
+    bundle,
+    vatParameters,
+    virtualObjectCacheSize,
+  ) {
+    /** @type { (item: Tagged) => unknown } */
+    function doSyscall(vatSyscallObject) {
+      return JSON.parse(
+        decoder.decode(
+          // @ts-ignore
+          // eslint-disable-next-line no-undef
+          issueCommand(Item.encode(['syscall', ...vatSyscallObject])),
+        ),
+      );
+    }
+
+    const syscall = harden({
+      send: (...args) => doSyscall(['send', ...args]),
+      callNow: (...args) => doSyscall(['callNow', ...args]),
+      subscribe: (...args) => doSyscall(['subscribe', ...args]),
+      fulfillToData: (...args) => doSyscall(['fulfillToData', ...args]),
+      fulfillToPresence: (...args) => doSyscall(['fulfillToPresence', ...args]),
+      reject: (...args) => doSyscall(['reject', ...args]),
+    });
+
+    const vatPowers = {
+      Remotable,
+      getInterfaceOf,
+      makeMarshal,
+      testLog,
+    };
+
+    const ls = makeLiveSlots(
+      syscall,
+      vatID,
+      vatPowers,
+      vatParameters,
+      // @ts-ignore  TODO: defend against non-numeric?
+      virtualObjectCacheSize,
+      // TODO: lsgc? API drift?
```

Comment:
Yeah, as I figure out more of the GC story, I'm using this argument to provide `WeakRef` and `FinalizationRegistry` to the liveslots code (since these are supposed to be stripped from all non-start-compartments), but the definition doesn't actually use it yet. If it's convenient, pass in the same things that the other supervisors do, so there will be less work to do once I get to update `makeLiveSlots` to use it.

Reply:
I guess it's not convenient. :) That is: tsc complains about the extra arg and I'm always nervous about untested code.